src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -421,32 +421,19 @@ private DotnetHost DotnetHostInfo
                     var commandLine = commandLineBuilder.ToString();
 
                     // ToolExe delegates back to ToolName if the override is not
-                    // set.
-                    // So, we can't check if it's unset, as that will recur
-                    // and stackoverflow.
-                    // However, checking only ToolPath is inadequate, as some
-                    // callers only set ToolExe, and not ToolPath (e.g. CLI).
-                    // So, do the check after _dotnetHostInfo is assigned, and
-                    // if ToolExe routes back here and returns
-                    // _dotnetHostInfo.ToolName, we know that ToolExe is unset.
-                    // So, if it does *not* return such, we know ToolExe is
-                    // explicitly overriden - so swap out the DotnetHost with
-                    // the passthrough implementation, as otherwise the command
-                    // line would be incorrect (it would have csc.dll in the
-                    // arguments).
-                    if (string.IsNullOrEmpty(ToolPath))
+                    // set.  So, if ToolExe == ToolName, we know ToolExe is not
+                    // explicitly overriden.  So, if both ToolPath is unset and
+                    // ToolExe == ToolName, we know nothing is overridden, and
+                    // we can use our own csc.
+                    if (string.IsNullOrEmpty(ToolPath) && ToolExe == ToolName)
```

Comment:
Same feedback.

Reply:
This particular conditional is duplicated between ManagedCompiler and InteractiveCompiler. Consider factoring into a common method. It is pretty small so may not be worth doing.